ccl/changefeedccl: TestCDCQuerySelectSingleRow failed

ccl/changefeedccl.TestCDCQuerySelectSingleRow [failed](https://tanzanite.cluster.engflow.com/invocations/default/15a84042-e21e-4cc6-8f60-c877b1af3525?testReportRun=3&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on release-24.1 @ [3a08f64f355ad645050a94e300fb50b47f8cec47](https://github.com/cockroachdb/cockroach/commits/3a08f64f355ad645050a94e300fb50b47f8cec47):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestCDCQuerySelectSingleRow (14m6s)
	TestCDCQuerySelectSingleRow/kafka (14m6s)

```
Stack: 

```
goroutine 119937 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* 
* goroutine 102628 [select]:
* github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs.func1()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:491 +0x14d
* created by github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs in goroutine 102605
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:462 +0x4f
* 
* goroutine 102768 [select]:
* github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs.func1()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:491 +0x14d
* created by github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs in goroutine 102605
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:462 +0x4f
* 
* goroutine 102771 [select, 2 minutes]:
* github.com/cockroachdb/cockroach/pkg/util/admission.makeStoreWorkQueue.func1()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:2251 +0x85
* created by github.com/cockroachdb/cockroach/pkg/util/admission.makeStoreWorkQueue in goroutine 102605
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:2248 +0x46c
* 
* goroutine 102606 [select]:
* github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFS).startTickerLocked.func1()
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:660 +0xf8
* created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFS).startTickerLocked in goroutine 102605
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:649 +0x6e
* 
* goroutine 102629 [select]:
* github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue.func2()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:414 +0x85
* created by github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue in goroutine 102605
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:411 +0x3be
* 
* goroutine 102641 [select]:
* github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker.func1()
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:243 +0xcb
* created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker in goroutine 102605
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:238 +0x58
* 
* goroutine 102638 [chan receive, 2 minutes]:
* github.com/cockroachdb/pebble.(*cleanupManager).mainLoop(0xc006774be0)
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:161 +0xe5
* github.com/cockroachdb/pebble.openCleanupManager.func1.1({0x8419138?, 0xc0053f21e0?})
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:96 +0x17
* runtime/pprof.Do({0x8418b78?, 0xca71840?}, {{0xc000f97740?, 0xc008c96fd0?, 0x4a823d?}}, 0xc008c96fc0)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x9d
* github.com/cockroachdb/pebble.openCleanupManager.func1()
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:95 +0x56
* created by github.com/cockroachdb/pebble.openCleanupManager in goroutine 102605
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:94 +0x265
* 
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCDCQuerySelectSingleRow.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-49005

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -11025,7 +11025,11 @@ func TestCDCQuerySelectSingleRow(t *testing.T) {
 		}
 		cfKnobs := knobs.DistSQL.(*execinfra.TestingKnobs).Changefeed.(*TestingKnobs)
 		cfKnobs.HandleDistChangefeedError = func(err error) error {
-			errCh <- err
+			// Only capture the first error -- that's enough for the test.
+			select {
+			case errCh <- err:
+			default:
+			}
 			return err
 		}
 	}

